### PR TITLE
Use lower-case headers consistently

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- Use lower-case headers consistently

## Why are we making these changes?

Consistently use lower-case headers in waiter codebase to avoid confusion in performing gets from headers.

### CODE REVIEW

File: waiter/docs/how-to-waiter.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Check out [Waiter Checklist](service-checklist.md) before you try running your s
 
 # Overview
 
-There are two ways to launch services on Waiter: the On-the-Fly API and the Token API.  The On-the-Fly API automatically starts HTTP services using custom HTTP headers coming from the client.  Alternatively, the Token API allows service maintainers to predefine a service.  Token-based services are reachable by HTTP clients using a DNS name or using a custom HTTP header, `X-Waiter-Token`.
```
Comment (nsinkov) [inline waiter/docs/how-to-waiter.md]: what about editing waiter/docs/parameters.md and waiter/docs/rest-api.md

File: waiter/integration/waiter/token_request_test.clj [hunk 2]
```diff
@@ -917,27 +917,27 @@
         (assert-response-status response 400)))
 
     (testing "can't use invalid token that's valid for zookeeper"
-      (let [response (make-request waiter-url "/pathabc" :headers {"X-Waiter-Token" "bad#token"})]
```
Comment (nsinkov) [inline waiter/integration/waiter/token_request_test.clj]: also edit waiter/integration/waiter/busy_instance_test.clj and waiter/integration/waiter/streaming_test.clj?

just run `waiter$ grep -r X-Waiter .` ?

**Comment by nsinkov** [comment]
maybe for every change, do a top-level `grep -r <string> .` to see that another upper-case instance still exists

File: waiter/docs/how-to-waiter.md (hunk 1 above)
Comment (shamsimam) [reply waiter/docs/how-to-waiter.md]: Too many to edit, so left them.

File: waiter/integration/waiter/token_request_test.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/integration/waiter/token_request_test.clj]: Done.

File: waiter/docs/how-to-waiter.md (hunk 1 above)
Comment (shamsimam) [reply waiter/docs/how-to-waiter.md]: Done now.

**Comment by shamsimam** [comment]
@nsinkov ready for another round of review.

**Pull request merged**